Documentation/General/emergentArchitectureDesign/emergentArchitectureDesign.tex

```diff
@@ -135,6 +143,16 @@ \subsubsection{REST API}
 This allows us to use JSON in the frontend, which is ideal in \Gls{JS}, but use classes and objects in Java, which greatly improves the testability of the responses.
 The backend will be tested with unit tests written using JUnit and Mockito.
 
+\subsubsection{SoundCloud integration}
+Moodcat uses the SoundCloud API for user login, metadata querying and, in the frontend, for streaming.
+To keep track of user's points, progress and stats Moodcat lets their users sign in with their SoundCloud account,
+when users press the login button they are redirected to the SoundCloud login page, which will, on succesfull login,
+provide a SoundCloud user id. This id is used to store and retrieve the statistics of users in our database.
+
+To fill the database with songs we have created a bulkInsert helper program. This will read a list of ids of the 
```

Comment:
Don't we want to split up the word blukInsert. It's now the same as in our code, but for someone who hasn't seen that it can be a bit vaue.  

I would suggest "To fill the database with songs we have created a helper program. This program uses our own list of the 1000 most populair tracks on SoundCloud. We fetch their metadata ....."